### ISSUE TITLE
nginx config broken in 22.1.9 update

### ISSUE THREAD

**Issue by rveety** [opnsense/plugins#3015]
After upgrading from 22.1.8_1 to 22.1.9 nginx will not start, logging this:

2022/06/23 10:54:00 [emerg] 93236#100289: unknown directive "js_include" in /usr/local/etc/nginx/nginx.conf:40

It looks like js_include is deprecated in this version (http://nginx.org/en/docs/njs/changes.html#njs0.7.1).  For now, I commented it out that line and manually started nginx and it works without the check_intercept functionality.

**Comment by fichtner** [comment]
Someone needs to fix plugin, but actually the nginx update caused it. For a workaround:

    # opnsense-revert -r 22.1.8 nginx

**Comment by degewo** [comment]
- [X] I have read the contributing guide lines at https://github.com/opnsense/plugins/blob/master/CONTRIBUTING.md
- [X] I have searched the existing issues, open and closed, and I'm convinced that mine is new.
- [X] The title contains the plugin to which this issue belongs

**Describe the bug**
Update from opnsense 21.1.8 to 21.1.9 with nginx 1.20.2_9,2 to 1.22.0_5,2 and plugin os-nginx version 1.28 produce broken config with error:

``nginx: [emerg] unknown directive "js_include" in /usr/local/etc/nginx/nginx.conf:40``

Revert like @fichtner mentioned is a possible workaround.

**To Reproduce**
Update from opnsense 21.1.8 to 21.1.9 with nginx 1.20.2_9,2.
Steps to reproduce the behavior:
1. Go to 'Lobby: Dashboard'
2. Click on 'Click to check for updates.'
3. Apply updates and reboot
4. nginx isn't starting

**Expected behavior**
Nginx starts with new nginx version 1.22.0_5,2 and plugin os-nginx version 1.28.

**Relevant log files**
1. Go to 'Services: Nginx: Logs / Global Error'
2. Filter Severity with 'emerg'
3. Result with 'unknown directive "js_include" in /usr/local/etc/nginx/nginx.conf:XX' should appear

**Additional context**
Like @rveety mentioned see (http://nginx.org/en/docs/njs/changes.html#njs0.7.1)[http://nginx.org/en/docs/njs/changes.html#njs0.7.1], 'js_include' is not supported anymore.

**Environment**
OPNsense 22.1.9-amd64
FreeBSD 13.0-STABLE
OpenSSL 1.1.1o 3 May 2022

Intel(R) Xeon(R) Bronze 3204 CPU @ 1.90GHz (6 cores, 6 threads)

**Comment by Arnavion** [comment]
IIUC, the fix is:

```diff
-js_include /usr/local/opnsense/scripts/nginx/ngx_functions.js;
-js_set $tls_intercepted check_intercept;
+js_import opnsense_ngx_functions from /usr/local/opnsense/scripts/nginx/ngx_functions.js;
+js_set $tls_intercepted opnsense_ngx_functions.check_intercept;
```

and in `/usr/local/opnsense/scripts/nginx/ngx_functions.js` :

```diff,js
-function create_secure_link(r) {
+export function create_secure_link(r) {
```

**Comment by johnnyslee** [comment]
The above fix seems working with no problem.
```diff
--- /usr/local/opnsense/service/templates/OPNsense/Nginx/http.conf.orig	2022-06-22 20:19:58.000000000 +0800
+++ /usr/local/opnsense/service/templates/OPNsense/Nginx/http.conf	2022-06-24 04:48:08.780768000 +0800
@@ -21,8 +21,8 @@
 
 #tcp_nopush     on;
 # https intercept detection
-js_include /usr/local/opnsense/scripts/nginx/ngx_functions.js;
-js_set $tls_intercepted check_intercept;
+js_import opnsense_ngx_functions from /usr/local/opnsense/scripts/nginx/ngx_functions.js;
+js_set $tls_intercepted opnsense_ngx_functions.check_intercept;
 
 # 200M should be big enough for file servers etc.
 client_max_body_size 200M;
```
```diff
--- /usr/local/opnsense/scripts/nginx/ngx_functions.js.orig	2022-06-22 20:19:58.000000000 +0800
+++ /usr/local/opnsense/scripts/nginx/ngx_functions.js	2022-06-24 04:48:08.785901000 +0800
@@ -47,3 +47,5 @@
     }
     return tls_result.status;
 }
+
+export default {check_intercept}
```

**Comment by sowoi** [comment]
i don't want to complain about a product that was given to me for free, but the bug would have been noticed before rolling it out with a simple "nginx -t". this is more than unprofessional.

**Comment by fichtner** [comment]
I can see this harsh conclusion for lack of release engineering internals. Cheers!

**Comment by Starkstromkonsument** [comment]
fixed by #3016

**Comment by fichtner** [comment]
Indeed, closing.
